sources/packages/forms/src/form-definitions/educationprogram.json

```diff
@@ -1,5871 +1,6462 @@
+                "multiple": false,
+                "unique": false
+            },
+            "allowCalculateOverride": false,
+            "encrypted": false,
+            "showCharCount": false,
+            "showWordCount": false,
+            "allowMultipleMasks": false,
+            "id": "ee9wfci",
+            "addons": []
         },
         {
-          "label": "Program description",
-          "autoExpand": false,
-          "showCharCount": true,
-          "tableView": true,
-          "validate": {
-            "customMessage": "Too Long",
-            "maxLength": 500,
-            "required": false,
-            "custom": "",
-            "customPrivate": false,
-            "strictDateValidation": false,
+            "title": "Panel",
+            "theme": "default",
+            "tooltip": "",
+            "customClass": "",
+            "collapsible": false,
+            "hidden": false,
+            "hideLabel": true,
+            "disabled": false,
+            "modalEdit": false,
+            "key": "panel",
+            "tags": [],
+            "properties": {},
+            "customConditional": "",
+            "conditional": {
+                "json": "",
+                "show": null,
+                "when": null,
+                "eq": ""
+            },
+            "logic": [],
+            "attributes": {},
+            "overlay": {
+                "style": "",
+                "page": "",
+                "left": "",
+                "top": "",
+                "width": "",
+                "height": ""
+            },
+            "type": "panel",
+            "label": "Panel",
+            "breadcrumb": "default",
+            "tabindex": "",
+            "input": false,
+            "tableView": false,
+            "components": [
+                {
+                    "label": "Level Of Study Codes",
+                    "labelWidth": "",
+                    "labelMargin": "",
+                    "customClass": "",
+                    "modalEdit": false,
+                    "defaultValue": "levelOfStudyCodes",
+                    "persistent": true,
+                    "protected": false,
+                    "dbIndex": false,
+                    "encrypted": false,
+                    "redrawOn": "",
+                    "customDefaultValue": "",
+                    "calculateValue": "value = [\r\n{credentialType: \"undergraduateCertificate\", levelOfStudyCode: \"8\"},\r\n{credentialType: \"undergraduateDiploma\", levelOfStudyCode: \"7\"},\r\n{credentialType: \"undergraduateDegree\", levelOfStudyCode: \"3\"},\r\n{credentialType: \"graduateCertificate\", levelOfStudyCode: \"5\"},\r\n{credentialType: \"graduateDiploma\", levelOfStudyCode: \"5\"},\r\n{credentialType: \"graduateDegreeOrMasters\", levelOfStudyCode: \"5\"},\r\n{credentialType: \"postGraduateOrDoctorate\", levelOfStudyCode: \"6\"},\r\n{credentialType: \"qualifyingStudies\", levelOfStudyCode: \"1\"}\r\n];",
```

Comment:
This can be converted to an object and the property value can be directly accessed without the need of using a `filter` by the consumer code.
```js
value = {
  undergraduateCertificate: 8,
  undergraduateDiploma: 7,
  undergraduateDegree: 3,
  graduateCertificate: 5,
  graduateDiploma: 5,
  graduateDegreeOrMasters: 5,
  postGraduateOrDoctorate: 6,
  qualifyingStudies: 1,
};
```
Then the consumer can access like `theObject["undergraduateCertificate"]`.

Reply:
Updated.

Reply:
I saw the code update, thanks for that, but do we need to declare all the property names as strings? Was it not possible to implement in the way that the sample was provided?